### ISSUE TITLE
incomplete search result

### ISSUE THREAD

**Issue by XinYaanZyoy** [KosmosisDire/obsidian-webpage-export#356]
**Describe the bug**
searching doesn't return the complete result, the one we get in the obsidian.

**Operating System**
Ubuntu 22.x

**To Reproduce**
1. export
2. serve
3. search on site 
4. search the same keyword on obsidian, and compare

**Screenshots**
will provide if necessary 

### IMPORTANT: Error Log
not relevant ig

**Comment by KosmosisDire** [comment]
I'm not sure what you mean, a screenshot would help

**Comment by XinYaanZyoy** [comment]
![image](https://github.com/KosmosisDire/obsidian-webpage-export/assets/32465479/03bd3191-f369-4742-9db0-9be792d6fb5e)
![image](https://github.com/KosmosisDire/obsidian-webpage-export/assets/32465479/42b315e1-6235-479b-b3e9-4a92ac1ade12)

"topos" in the obsidian has 207 results, but if just count the files it's >15, but in the site it's only 5.

**Comment by KosmosisDire** [comment]
I think what @7Mik means is that I can't tell if there is a document in obsidan's search results that actually contains the word "topos" but isn't shown in the document. I am not 100% sure how obsidian's search works, but does every file in that list contain the word topos inside of it?

If so then what you're saying is that theoretically every document in that list should show in the website too because all of them have an exact match with "topos"?

If not all of them have an exact match then I think the current behavior is expected.

**Comment by XinYaanZyoy** [comment]
yes, "every file in that list contain the word topos inside of it".

yes, that was my expectation.

as far as my understanding goes, those "non-exact-matching entries" are not shown in the search bar function (the left top thing), but are in the find or open bar function (the ctrl+o center thing). that is, all of the results have exact match. if the plugin is trying to mimic the first function, I expect the same results on the plugin, though I'm completely fine and more than satisfied with the current implemented function, so no worries, I created the issue hoping that it'll help the development in some sense.

**Comment by KosmosisDire** [comment]
So I think some files that obsidian is showing actually don't have topos inside of them. For example: https://xinyaanzyoy.github.io/wiki/polynomial-functors-a-general-theory-of-interaction-by-spivak-and-niu.html

However I was able to find some files that are not shown in the webpage search, that are shown correctly in obsidian.
I think this is actually working as intended, however the webpage search thinks those documents have low priority matches. Therefore it chooses to show only the highest quality matches.

I can adjust it to show more matches at the cost of possibly showing less relevant matches. Or I could consider adding an option to adjust the filter's sensitivity. Thank you for the report :)

**Comment by XinYaanZyoy** [comment]
the content of the file DOES have the searched word, in this case, it is in the link href.

**Comment by KosmosisDire** [comment]
Ahhh, I see! The indexer throws out all the hrefs because I figure usually people wouldn't want to search those. But I suppose now that you point it out you may want to search for documents in which you linked to a website with a certain name. So I will consider loosening the search terms as well as including hrefs into the search index. Thanks!

**Comment by KosmosisDire** [comment]
This should be improved in beta 1.8.1-0b, let me know if it is. Thanks!
